api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/HttpSender.java

```diff
@@ -0,0 +1,98 @@
+import java.net.URL;
+
+public class HttpSender implements Hl7v2Sender {
+
+    private final Log log = LogFactory.getLog(this.getClass());
+
+    @Autowired
+    private RegistrationCoreProperties config;
+
+    @Override
+    public Message sendPdqMessage(Message request) throws LLPException, IOException, HL7Exception {
+        URL url = new URL(config.getPdqEndpoint());
+        return sendMessage(request, url);
+    }
+
+    @Override
+    public Message sendPixMessage(Message request) throws LLPException, IOException, HL7Exception {
+        URL url = new URL(config.getPixEndpoint());
+        return sendMessage(request, url);
+    }
+
+    private Message sendMessage(Message request, URL url) throws HL7Exception, LLPException, IOException {
+
+        PipeParser parser = new PipeParser();
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Content-Type", "application/hl7-v2+er7; charset=utf-8");
+        String authentication = config.getMpiUsername() + ":" + config.getMpiPassword();
+        String encoded = "Basic "+ Base64.encode(authentication.getBytes());
+        connection.setRequestProperty("Authorization", encoded);
+        connection.setRequestMethod("POST");
+        connection.setDoOutput(true);
+
+        try
+        {
+            if(log.isDebugEnabled())
+                log.debug(String.format("Sending to %s : %s", url, parser.encode(request)));
+
+            String message = "\u000B" + request.toString() + "\u001C" + "\r";
+            connection.getOutputStream().write(message.getBytes("UTF-8"));
+
+            int code = connection.getResponseCode();
+            Message response = null;
+
+            if (!isSuccessfulResponseCode(code)) {
+                throw new MpiException(String.format("MPI connection error. "
+                                + "Response code: %s, response message: %s, error stream: %n%s ",
+                        code,
+                        connection.getResponseMessage(),
+                        IOUtils.toString(connection.getErrorStream())
+                ));
+            }
+            response = getResponse(connection, parser);
+    
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Response from %s : %s", url, parser.encode(response)));
+            }
+            
+            return response;
+        }
+        finally
+        {
+            connection.disconnect();
+        }
+    }
+
+    private Message getResponse(HttpURLConnection connection,  PipeParser parser) throws HL7Exception, LLPException, IOException
+    {
+        InputStream inputStr = connection.getInputStream();
+        String encoding = connection.getContentEncoding() == null ? "UTF-8"
+                : connection.getContentEncoding();
+        String decoded = IOUtils.toString(inputStr, encoding);
+        decoded = decoded.substring(1, decoded.length() - 1);
```

Comment:
Would be nice to add some informative comments to explain why this is necessary.

Reply:
Can you address this please?

Reply:
Honestly I am not sure... I didn't write this part. Michal Kus added it in this commit https://github.com/IsantePlus/openmrs-module-registrationcore/commit/a88b25411eee5dd980b58d000a21a1ce09ce568e